### PR TITLE
GEOMESA-1143 within writeLineString method, Fixed the cast exception …

### PR DESCRIPTION
**Pull request by mbeebe81**
I added back writeMultiLineString method.  But I only use that if the LineString intersects and creates a MultiLineString.  I added an additional pattern match there which may be overkill.  Can a MultiLineString contain other geometries besides lineStrings (like a point or multiPoint)?

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/DensityIterator.scala [hunk 1]
```diff
@@ -75,9 +75,23 @@ class DensityIterator extends KryoLazyDensityIterator with LazyLogging {
...
+      case geom1: Geometry              => writeNonPoint(geom1, weight, result)
+    }
+  }
+
+  /**
+    * Intersection of line and geoHash can form a multiLineString */
+  private def writeMultiLineString(geom: MultiLineString, weight: Double, result: DensityResult): Unit = {
+      (0 until geom.getNumGeometries).foreach{ i => geom.getGeometryN(i) match {
+        case g: LineString => writeLineString(g, weight, result)
+        case g: Geometry   => writeNonPoint(g, weight, result)
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/DensityIterator.scala]: I'm pretty sure a multi line string couldn't have anything other than a line string for children.

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/DensityIterator.scala [hunk 2]
```diff
@@ -75,9 +75,23 @@ class DensityIterator extends KryoLazyDensityIterator with LazyLogging {
...
+      case geom1: MultiLineString       => writeMultiLineString(geom1, weight, result)
+      case geom1: Geometry              => writeNonPoint(geom1, weight, result)
+    }
+  }
+
+  /**
+    * Intersection of line and geoHash can form a multiLineString */
+  private def writeMultiLineString(geom: MultiLineString, weight: Double, result: DensityResult): Unit = {
+      (0 until geom.getNumGeometries).foreach{ i => geom.getGeometryN(i) match {
+        case g: LineString => writeLineString(g, weight, result)
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/DensityIterator.scala]: this is going to do another intersection with the geohash - maybe we could pull out the getCoords().sliding etc snippet and call that directly here.

**Comment by elahrvivaz** [comment]
@mbeebe81 looks good

**Comment by elahrvivaz** [comment]
@mbeebe81 did you sign the eclipse CLA as outlined here? https://github.com/locationtech/geomesa/blob/master/CONTRIBUTING.md#contributor-license-agreement
the eclipse validation plugin is failing to validate you.

**Comment by mbeebe81** [comment]
I did fill out the contributing.md form.  I used my mbeebe@ccri.com email address which is my primary email address. I filled it out today.

**Comment by elahrvivaz** [comment]
looks good, merging

**Comment by elahrvivaz** [comment]
merged as 717b83e2e52f8827d192347be6766f343afbcdc7

**Pull request closed without merging**